lib/SPIRVProducerPass.cpp

```diff
@@ -83,6 +83,9 @@ cl::opt<bool>
     ShowProducerIR("show-producer-ir", cl::init(false), cl::ReallyHidden,
                    cl::desc("Dump the IR at the start of SPIRVProducer"));
 
+cl::opt<std::string> SPIRVVersionOpt("spirv-std", cl::init("1.0"), cl::Hidden,
```

Comment:
Ok, I had it marked as hidden but will make it inferred since it is a translation tool.

I am sure there are other problems (I noted that the entry point interfaces are common for all kernels) but it is encouraging that spirv-val passed for all tests when defaulting to SPIRV 1.5.

Reply:
Actually, thinking more about this. Shouldn't the version be driven by the clvk driver based on the underlying vulkan implementation version? Currently, clvk is hard coded to SPIRV 1.0 but it does query the vk version from the underlying driver.

Reply:
clvk will catch the problem when it looks module's capabilities. clspv doesn't have useful error handling except in the frontend plugin right now. I'd prefer not to hit llvm_unreachable because you're on a 1.0 implementation and used subgroups. If you can check everything in the frontend, it might be ok.